pwiz_tools/Skyline/Alerts/ShareResultsFilesDlg.cs

```diff
@@ -0,0 +1,452 @@
+
+            // Set initial directory to default if not applicable
+            var initialDir = Path.GetDirectoryName(docPath) ?? Settings.Default.SrmResultsDirectory;
+            if (string.IsNullOrEmpty(initialDir))
+                initialDir = null;
+
+            // Set the initial directory to the current working directory if present
+            searchFolderDialog.SelectedPath = initialDir;
+
+            searchFolderDialog.Description = $"Please select the folder containing the missing files"; // Description
+
+            if (searchFolderDialog.ShowDialog() == DialogResult.OK)
+            {
+                SearchDirectoryForMissingFiles(searchFolderDialog.SelectedPath);
+            }
+        }
+
+        /// <summary>
+        /// Preforms search for missing items within directory given directory path. Split
+        /// for testing purposes and incompatibility with form types.
+        /// </summary>
+        /// <param name="folderPath"></param>
+        public void SearchDirectoryForMissingFiles(string folderPath)
+        {
+            // Get all file/directory information from given folder path
+            var files = Directory.GetFiles(folderPath);
+            var folders = Directory.GetDirectories(folderPath);
+            var set = new HashSet<string>();
+
+            // Add all file paths in folder to hash set
+            foreach (var file in files)
+            {
+                set.Add(file);
+            }
+
+            // Add all folder paths in folder to hash set
+            foreach (var folder in folders)
+            {
+                set.Add(folder);
+            }
+
+            set.Remove(null);
+
+            // Check if the folder contained any missing items
+            foreach (var rawFiles in set)
+            {
+                if (missingListBox.Items.Contains(Path.GetDirectoryName(rawFiles) != null) || missingListBox.Items.Contains(Path.GetFileName(rawFiles)))
+                {
+                    checkedListBox.Items.Add(rawFiles, true);
+                    missingListBox.Items.Remove(Path.GetDirectoryName(rawFiles) != null);
+                    missingListBox.Items.Remove(Path.GetFileName(rawFiles));
+                }
+            }
+            checkedStatus.Text = UpdateLabel();
+        }
+
+
+        /// <summary>
+        /// All list box information used to store the state of the from
+        /// </summary>
+        public class AuxiliaryFiles
+        {
+            public List<CheckedListBoxItems> _checkBoxFiles;
+            public List<string> _missingCheckBoxFiles;
+
+            public AuxiliaryFiles(List<CheckedListBoxItems> checkedFiles, List<string> missingFiles)
+            {
+                _checkBoxFiles = checkedFiles;
+                _missingCheckBoxFiles = missingFiles;
+            }
+        }
+
+        /// <summary>
+        /// Checked list box information
+        /// </summary>
+        public class CheckedListBoxItems
```

Comment:
I think this class should not be plural and should be named "CheckedListBoxItem".